{{cookiecutter.project_repo}}/.github/workflows/package.on-dispatch.yml

```diff
@@ -0,0 +1,88 @@
+name: Publish Python from Workflow Dispatch
+# Publishes a Python Package with a specified git tag to Azure DevOps, and creates a release
+
+on:
+  workflow_dispatch:
+    inputs:
+      poetry-version:
+        description: 'Version to bump. Can be major, minor, patch, prerelease or a valid semver string.'
+        required: true
+        default: 'prerelease'
+      description-body:
+        description: ''
+        required: false
+        default: ''
+
+jobs:
+  deploy-tag:
+    runs-on: ubuntu-latest
+    environment: AzDoUKHOGETUpload
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Get full Python version
+      id: full-python-version
+      run: |
+        echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
+
+    - name: Install Dependencies
+      run: |
+        sudo snap install dotnet-sdk --classic
+        sudo snap alias dotnet-sdk.dotnet dotnet
+        sudo add-apt-repository -y ppa:ubuntugis/ppa
+        sudo apt-get update
+        sudo apt-get install -y --no-install-recommends libreadline-gplv2-dev libncursesw5-dev libssl-dev libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev
+        sudo apt-get install -y --no-install-recommends libffi-dev
+        sudo apt-get install -y --no-install-recommends lzma liblzma-dev
+        sudo apt-get install -y --no-install-recommends libgeos-dev libgeos++-dev
+        sudo apt-get install -y --no-install-recommends libproj-dev proj-data proj-bin
+        sudo apt-get install -y --no-install-recommends libsnappy-dev
+        sudo apt-get install -y --no-install-recommends python3-h5py
+        sudo apt-get install -y --no-install-recommends gdal-bin libgdal-dev python3-gdal
+        python -m pip install --upgrade pip poetry
+
+    - name: Bump version
+      run : |
+        poetry version ${% raw %}{{ {% endraw %}github.event.inputs.poetry-version }} {% endraw %}
+
+    - name: Create git tag
+      run : |
+        git tag $(poetry version --short)
+        git push origin $(poetry version --short)
+
+    - name: Publish to package repository
+      env:
+        DOTNET_ROOT: /snap/dotnet-sdk/current
+        POETRY_PYPI_TOKEN_UKHOGET: ${% raw %}{{ {% endraw %}secrets.UKHO_{{cookiecutter.project_slug}}_AZDO_UKHOGET_UPLOAD_PAT {% raw %} }} {% endraw %}
```

Comment:
The slug may need to be converted to uppercase

Reply:
I think this is done, uncertain best way to test cookiecutter (I can just create a sample project if there's no better way)

Reply:
Confirmed, now coerces to uppercase